### PR TITLE
Added entries interface

### PR DESCRIPTION
**Pull request by aviezerl**

### CODE REVIEW

**Comment by ramijaschek** [comment]
A. The fact we are doing a reload on each get seems a bit excessive - maybe just ask for a flag to do that? Basically we are expecting post_etl to write those. Also - if you are doing a reload each time - why would we actively ask for a reload ? I understand that the reload stops with the timestamp - but it's still a bit of time.
B. I see you are allowing a write to all dbs - but then a get_all is always from a single db (even if I did not write one) - why? 

Bottom line - two things we need clarity on:
A. Are we assuming checking timestamp is no time at all - and then no need for reload.
B. Are we allowing work on multiple directories - if yes - db_dir should be allowed to be a list or null to mean all.

**Comment by aviezerl** [comment]
A. We are indeed assuming the timestamp check is negligent. This is how naryn currently works - for every query of the database, it checks the timestamp of .naryn and reloads it if needed. I think the confusion comes from using the verb 'reload', which now only loads a small yaml file in contrast to the `emr_db.reload()` which not only re-reads the .naryn file but also scans the disk for new track files. Maybe we should change the name to `emr_db.refresh()`? What do you think? 

B. We are allowing to write to another db explicitly, but indeed this is not the regular use case, and therefore the default behavior in all the functions is to refer to only the first db. Why should `emr_db.get_all()` be different in that respect?

**Comment by ramijaschek** [comment]
A. OK - but if we are doing an auto-reload on each access - why do we need a 'reload'/'refresh' ?
B. It's not critical - but currently the only way to get all annotations is to iterate through all directories explicitly - doesn't sound reasonable.

**Comment by aviezerl** [comment]
> A. OK - but if we are doing an auto-reload on each access - why do we need a 'reload'/'refresh' ? 

We need it for the rare cases that you want something to be updated now, for example, when you know you are going to access the file in parallel, and you prefer not to reload it from the disk every time. I agree this is an advanced feature, but we might need it in rare cases, even if the loading time is negligent. 

> B. It's not critical - but currently the only way to get all annotations is to iterate through all directories explicitly - doesn't sound reasonable.

I don't know. On one hand, you are right that this would be very cumbersome, but on the other hand, I do not want to encourage the user to use entries from more than one database at the same time. Think about time-scope for example: what happens when there are different parameters in different databases the user is connected to? I think we need to advocate for a single source of truth in such cases - only the first database, and therefore the fact that other databases may have other parameters shouldn't be exposed easily to the user. 
I can add a separate function for loading all the entries from all databases (`emr_entries.get_all_dbs()`?), but I am afraid this may lead the user (a.k.a us) to do things we will regret afterward. What do you think?

**Comment by ramijaschek** [comment]
I assumed the entries would be only in the main DB - so I did not really look for files in the other directories. If you did decide to support files in all - I think it makes sense to accept a vector of directories to look at (and I'll put in .naryn$EMR_ROOTS if I want to scan all).

**Comment by aviezerl** [comment]
Good idea. I will support a vector of dbs in the `db_dir` parameter.

**Comment by aviezerl** [comment]
Done. Have a look when you have time, and we will merge.
